pkg/provider/azure_vmssflex.go

```diff
@@ -69,7 +72,13 @@ func newFlexScaleSet(az *Cloud) (VMSet, error) {
 // GetNodeNameByProviderID gets the node name by provider ID.
 func (fs *FlexScaleSet) GetNodeNameByProviderID(providerID string) (types.NodeName, error) {
 
-	return types.NodeName(""), nil
+	// NodeName is part of providerID for standard instances.
+	matches := providerIDRE.FindStringSubmatch(providerID)
```

Comment:
as suggested at https://github.com/kubernetes-sigs/cloud-provider-azure/pull/2155#discussion_r942371304, because nodename may be different from vmname, this needs to be queried from a cached mapping

Reply:
Make sense. I just removed the implementation of this function for now. I will add it back fter #2155 is merged